apps/api/src/conversations/conversations.gateway.ts

```diff
@@ -0,0 +1,292 @@
+        newConversation,
+        client.data as User,
+      );
+    if (newConversationMessage) {
+      this.server
+        .in(`user_${client.data.id}`)
+        .socketsJoin(`conversation_${conversation.id}`);
+      for (const participant of newConversation.participants) {
+        this.server
+          .in(`user_${participant}`)
+          .socketsJoin(`conversation_${conversation.id}`);
+      }
+      client
+        .to(`conversation_${conversation.id}`)
+        .emit('newConversation', instanceToPlain(conversation));
+    }
+    return conversation;
+  }
+
+  @SubscribeMessage('getUnread')
+  async unreadCount(
+    @ConnectedSocket() client: Socket,
+  ): Promise<unreadMessagesResponse> {
+    return await this.conversationsService.unreadCount(client.data as User);
+  }
+
+  @SubscribeMessage('getMessages')
+  async getMessages(
+    @ConnectedSocket() client: Socket,
+    @MessageBody() { id }: isUUIDDto,
+  ): Promise<Message[]> {
+    return await this.conversationsService.getMessages(client.data as User, id);
+  }
+
+  @SubscribeMessage('postMessage')
+  async postMessage(
+    @ConnectedSocket() client: Socket,
+    @MessageBody() { content }: sendMessageDto,
+    @MessageBody() { id }: isUUIDDto,
+  ): Promise<Message> {
+    const ret = await this.conversationsService.postMessage(
+      client.data as User,
+      id,
+      content,
+    );
+    if (ret)
+      client
+        .to(`conversation_${id}`)
+        .emit('newMessage', { id, message: instanceToPlain(ret) });
+    return ret;
+  }
+
+  @SubscribeMessage('joinConversation')
+  async joinConversation(
+    @ConnectedSocket() client: Socket,
+    @MessageBody() { id }: isUUIDDto,
+    @MessageBody('password') password: string | undefined,
+  ): Promise<Conversation> {
+    const { conversation, joinMessage } =
+      await this.conversationsService.joinConversation(
+        client.data as User,
+        id,
+        password ? password : null,
+      );
+    if (joinMessage) {
+      this.server
+        .in(`user_${client.data.id}`)
+        .socketsJoin(`conversation_${conversation.id}`);
+      client
+        .to(`user_${client.data.id}`)
+        .emit('newConversation', instanceToPlain(conversation));
+      this.server
+        .in(`conversation_${conversation.id}`)
+        .except(`user_${client.data.id}`)
+        .emit('userJoined', {
+          id,
+          user: instanceToPlain(
+            await this.usersService.getById(client.data.id),
+          ),
+        });
+      this.server
+        .in(`conversation_${conversation.id}`)
+        .except(`user_${client.data.id}`)
+        .emit('newMessage', { id, message: instanceToPlain(joinMessage) });
+    }
+    return conversation;
+  }
+
+  @SubscribeMessage('getParticipants')
+  async getConversationPartipants(
+    @ConnectedSocket() client: Socket,
+    @MessageBody() { id }: isUUIDDto,
+  ): Promise<ConversationRole[]> {
+    return await this.conversationsService.getConversationParticipants(
```

Comment:
I don't think that awaiting before returning is necessary here (https://eslint.org/docs/latest/rules/no-return-await :slightly_smiling_face: :slightly_smiling_face: :slightly_smiling_face: this is my favorite website, I consult it every after noon during my _goûter_ :slightly_smiling_face: :slightly_smiling_face:)

Reply:
Not exactly what it says, you're oversimplifying. It says that removes the promise form the call stack, which can be a problem when you are debugging. We are currently debugging. Removed it for now but might come back